src/portfft/committed_descriptor_impl.hpp

```diff
@@ -521,6 +553,78 @@ class committed_descriptor_impl {
     return dispatch<calculate_twiddles_struct>(level, dimension_data, kernels);
   }
 
+  /**
+   * Sets the specialization constants for the global implementation
+   * @param input_kernels_and_metadata vector of input_bundles_and_metadata_t
+   * @param num_forward_factors  Number of forward factors
+   * @param num_backward_factors Number of backward factors
+   * @param compute_direction direction of compute: forward / backward
+   * @param is_prime whether or not the dimension size is a prime number
+   * @param scale_factor Scaling factor to be applied to the result
+   */
+  void set_global_impl_spec_constants(std::vector<input_bundles_and_metadata_t>& input_kernels_and_metadata,
```

Comment:
Because of the way `set_spec_constant_struct` is set up. It only accepts one kernel bundle instead of accepting a vector, so you need one loop to determine the spec constants to be set which would be then passed to the struct. 

This is the same loop we had [here](https://github.com/codeplaysoftware/portFFT/blob/d5fe215de93e38244278909d5f2da8c5853e566f/src/portfft/committed_descriptor_impl.hpp#L554), just refactored out in a separate functions